zookeeper-server/src/main/java/org/apache/zookeeper/server/watch/WatcherCleaner.java

```diff
@@ -102,24 +104,24 @@ public void addDeadWatcher(int watcherBit) {
                 totalDeadWatchers.get() >= maxInProcessingDeadWatchers) {
             try {
                 RATE_LOGGER.rateLimitLog("Waiting for dead watchers cleaning");
-                synchronized(totalDeadWatchers) {
-                    totalDeadWatchers.wait(100);
-                }
-            } catch (InterruptedException e) {
-                LOG.info("Got interrupted while waiting for dead watches " +
-                        "queue size");
-            }
-        }
-        synchronized (this) {
-            if (deadWatchers.add(watcherBit)) {
-                totalDeadWatchers.incrementAndGet();
-                if (deadWatchers.size() >= watcherCleanThreshold) {
-                    synchronized (cleanEvent) {
-                        cleanEvent.notifyAll();
-                    }
-                }
-            }
+				synchronized (processingCompletedEvent) {
```

Comment:
@tumativ looks like we still have some indent problem for this patch, can you help correct those?

Reply:
We don't have a general format wiki yet, we'll work on that, in general it's 4 white space after { line.